test/cbmc/proofs/handleUnexpectedEvents/handleUnexpectedEvents_harness.c

```diff
@@ -0,0 +1,57 @@
+/*
+ * AWS IoT Over-the-air Update v3.1.0
+ * Copyright (C) 2021 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/**
+ * @file handleUnexpectedEvents_harness.c
+ * @brief Implements the proof harness for handleUnexpectedEvents function.
+ */
+/*  Ota Agent includes. */
+#include "ota.h"
+#include "stubs.h"
+#include <stdlib.h>
+
+extern OtaAgentContext_t otaAgent;
+extern void handleUnexpectedEvents( const OtaEventMsg_t * pEventMsg );
+
+void handleUnexpectedEvents_harness()
+{
+    OtaEventMsg_t * pEventMsg;
+    OtaEvent_t eventId;
+
+    pEventMsg = ( OtaEventMsg_t * ) malloc( sizeof( OtaEventMsg_t ) );
+
+    /* pEventMsg cannot be NULL as it is statically initialized in receiveAndProcessOtaEvent
+     * before handleUnexpectedEvents function is called. */
+    __CPROVER_assume( pEventMsg != NULL );
+
+    /* eventId can only assume values of OtaEvent_t enum. */
+    __CPROVER_assume( ( eventId >= OtaAgentEventStart ) && ( eventId <= OtaAgentEventMax ) );
+
+    pEventMsg->eventId = eventId;
+
+    /* Initialize OtaAppCallback to an empty callback function. */
+    otaAgent.OtaAppCallback = otaAppCallbackStub;
+
+    ( void ) handleUnexpectedEvents( pEventMsg );
```

Comment:
The results of the call to handleUnexpectedEvents with eventID = OtaAgentEventReceivedFileBlock is not validated.
https://github.com/aws/ota-for-aws-iot-embedded-sdk/blob/82bac55e867a971f68706f0c42d8780e5aa4b6bf/source/ota.c#L2796